### PR TITLE
Convert Boost to tsx

### PR DESCRIPTION
**Pull request by bjornstar**
There are 4 typescript errors that I don't know how to deal with:

```
src/effects/Boost.tsx:27:9 - error TS2531: Object is possibly 'null'.

27         ref.current.setMatrixAt(i + j, o.matrixWorld)
           ~~~~~~~~~~~

src/effects/Boost.tsx:28:9 - error TS2531: Object is possibly 'null'.

28         ref.current.instanceMatrix.needsUpdate = true
           ~~~~~~~~~~~

src/effects/Boost.tsx:34:37 - error TS2322: Type 'null' is not assignable to type 'BufferGeometry'.

34     <instancedMesh ref={ref} args={[null, null, length]}>
                                       ~~~~

src/effects/Boost.tsx:34:43 - error TS2322: Type 'null' is not assignable to type 'Material | Material[]'.

34     <instancedMesh ref={ref} args={[null, null, length]}>
                                             ~~~~
```

I figure I should ask for help first before I try more files.

### CODE REVIEW

File: src/effects/Boost.tsx [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
 const o = new Object3D()
 const boostPositions = [new Vector3(-0.4, -0.5, -1.8), new Vector3(0.4, -0.5, -1.8)]
 
 export function Boost({ opacity = 0.5, length = 12, size = 0.1 }) {
-  const ref = useRef()
-  let n
-  let j
-  let ctrl
-  let progress
+  const ref = useRef<InstancedMesh>(null)
```
Comment (Gusted) [inline src/effects/Boost.tsx:11]:
```suggestion
      const ref = useRef<InstancedMesh>(null!)
```

**Comment by bjornstar** [comment]
Ok, now we're left with:

```
src/effects/Boost.tsx:34:37 - error TS2322: Type 'null' is not assignable to type 'BufferGeometry'.

34     <instancedMesh ref={ref} args={[null, null, length]}>
                                       ~~~~

src/effects/Boost.tsx:34:43 - error TS2322: Type 'null' is not assignable to type 'Material | Material[]'.

34     <instancedMesh ref={ref} args={[null, null, length]}>
                                             ~~~~
```

**Comment by Gusted** [comment]
You're not going to like this one.

```tsx
<instancedMesh ref={ref} args={[null as unknown as BufferGeometry, null as unknown as Material, length]}>
```

You cant cast null to a defined type, but with a little MITM of `unknown` you can cast it as any type.

Or you could go the easy way and use
```tsx
<instancedMesh ref={ref} args={[null as any, null as any, length]}>
```

These type declarations of this TSX are are a bit incorrect as the `instancedMesh` will work without these `args` but still need the length.

**Comment by drcmda** [comment]
looks like TS types for instancedmesh are wrong here, they are most definitively optional. imesh inherits from mesh:

```jsx
Mesh( geometry : BufferGeometry, material : Material )
geometry — (optional) an instance of BufferGeometry. Default is a new BufferGeometry.
material — (optional) a single or an array of Material. Default is a new MeshBasicMaterial
```

@joshuaellis could we change these for the next @types/three release?

**Comment by joshuaellis** [comment]
@drcmda hm, we have to pass count, so null wouldn't suffice as part of the args call anyway, you'd have to replace it with `undefined` in the args array otherwise the defaults for `Mesh` wouldn't activate which is why it _works_.

Maybe @bjornstar you could open an issue in [`three-ts-types`](https://github.com/three-types/three-ts-types) and we can arrange this as a fix.

**Comment by bjornstar** [comment]
> @drcmda hm, we have to pass count, so null wouldn't suffice as part of the args call anyway, you'd have to replace it with `undefined` in the args array otherwise the defaults for `Mesh` wouldn't activate which is why it _works_.
> 
> Maybe @bjornstar you could open an issue in [`three-ts-types`](https://github.com/three-types/three-ts-types) and we can arrange this as a fix.

Sure, I've created an issue here: https://github.com/three-types/three-ts-types/issues/92

**Comment by joshuaellis** [comment]
I've opened an issue in `three.js` as you can see above. For now I would use `//@ts-expect-error` and put a link to the `three` issue.

**Comment by bjornstar** [comment]
> I've opened an issue in `three.js` as you can see above. For now I would use `//@ts-expect-error` and put a link to the `three` issue.

Excellent, thank you!

**Comment by bjornstar** [comment]
@drcmda @Gusted Thank you, this is exactly what I was hoping to see in the migration to typescript.

**Pull request merged**